var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -517,8 +549,14 @@ def cmake_args(self):
                 options.append("-DGMX_GPU:STRING=CUDA")
             elif "+opencl" in self.spec:
                 options.append("-DGMX_GPU:STRING=OpenCL")
-            elif "+sycl" in self.spec:
+            elif "+sycl" in self.spec or "+rocm" in self.spec:
                 options.append("-DGMX_GPU:STRING=SYCL")
+                if "+rocm" in self.spec:
+                    options.append("-DGMX_SYCL_HIPSYCL:BOOL=ON")
```

Comment:
That's how GROMACS 2023 works, but it is deprecated in GROMACS 2024 in favor of `-DGMX_SYCL=ACPP`. 

The `Dhipsycl_DIR` and `DHIPSYCL_TARGETS` should also be renamed when using GROMACS `@2024:` and AdaptiveCpp `@23.10.0:`.

We don't have to deal with it right now if we can tolerate CMake warnings.